### PR TITLE
Allow concat of repeat

### PR DESCRIPTION
**Pull request by domoritz**
Also fix size signals.

### CODE REVIEW

**Comment by domoritz** [comment]
Tests are passing.

File: src/compile/compile.ts [hunk 1]
```diff
@@ -96,15 +96,7 @@ export function assembleNestedMainGroup(model: Model) {
...
-  const hasLayout = !!model.assembleLayout();
-  const parentEncodeEntry = {
-    ...(!hasLayout ? {
-      width: {signal: 'width'},
-      height: {signal: 'height'},
-    } : {}),
-    ...model.assembleParentGroupProperties()
-  };
-
+  const parentEncodeEntry = model.assembleParentGroupProperties();
```
Comment (kanitw) [inline src/compile/compile.ts]: I think this is can be incorrect, if the root model has `name`, you won't correctly output `width` and `height` signals for this nested main group.

Comment (domoritz) [inline src/compile/compile.ts]: Hmm, can you give an example?

Comment (domoritz) [inline src/compile/compile.ts]: Okay, looks like it's already incorrect on master.

Comment (domoritz) [inline src/compile/compile.ts]: https://github.com/vega/vega-lite/issues/2326

File: examples/vg-specs/bar_grouped.vg.json [hunk 2]
```diff
@@ -264,20 +264,11 @@
...
-                            "height": {
-                                "signal": "child_height"
-                            },
                             "stroke": {
                                 "value": "#ccc"
                             },
                             "strokeWidth": {
                                 "value": 0
-                            },
-                            "fill": {
```
Comment (kanitw) [inline examples/vg-specs/bar_grouped.vg.json]: @domoritz Why removing fill in facet group?  I guess this might break brushing.

Comment (domoritz) [inline examples/vg-specs/bar_grouped.vg.json]: Any idea why this is happening? I don't see where I made a change that could cause this.

Comment (domoritz) [inline examples/vg-specs/bar_grouped.vg.json]: I tried it and it works.

Comment (kanitw) [inline examples/vg-specs/bar_grouped.vg.json]: What works?

File: src/compile/facet.ts [hunk 3]
```diff
@@ -332,6 +332,9 @@ function getFacetGroupProperties(model: FacetModel) {
   const encodeEntry = model.assembleParentGroupProperties();
 
   return {
+    width: model.child.getSizeSignalRef('width'),
```
Comment (kanitw) [inline src/compile/facet.ts]: I wonder if this is the right fix.  Shouldn't `model.assembleParentGroupProperties` already produce what we need?  

I wonder if there is something actually wrong with `hasSubPlotWithXy`? 
(As commented, I'm not sure if we really need it anyway.)

File: examples/vg-specs/bar_grouped.vg.json (hunk 2 above)
Comment (domoritz) [inline examples/vg-specs/bar_grouped.vg.json]: All interactive specs still work.

File: src/compile/facet.ts (hunk 3 above)
Comment (domoritz) [inline src/compile/facet.ts]: I wonder the same but I wanted to fix other things first.

File: examples/vg-specs/bar_grouped.vg.json (hunk 2 above)
Comment (arvind) [inline examples/vg-specs/bar_grouped.vg.json]: My vote would be to keep the transparent fills, unless there's a compelling reason to get rid of them. Having an undefined fill is a frequent source of bugs with capturing interaction events.

Comment (domoritz) [inline examples/vg-specs/bar_grouped.vg.json]: Okay, I wish I understood which line changed that... looking into it.

Comment (domoritz) [inline examples/vg-specs/bar_grouped.vg.json]: Fixed.

**Comment by arvind** [comment]
Thanks!
-- 
-Arvind
-----
Arvind Satyanarayan <http://arvindsatya.com>
Sent from my iPhone

**Pull request merged**